src/mcedit2/panels/worldmeta.py

```diff
@@ -0,0 +1,255 @@
+        self._toggleViewAction = callButton
+
+        load_ui("panels/worldmeta.ui", baseinstance=self)
+
+        self.worldNBTEditor.editorSession = self.editorSession
+        self.editorSession.revisionChanged.connect(self.revisionDidChange)
+
+        self.updateNBTTree()
+        self.setupPanel()
+        self.updatePanel()
+
+        centerWidgetInScreen(self)
+
+    def updatePanel(self):
+        self.defaultGamemodeCombo.setCurrentIndex(self.worldMeta.GameType)
+        self.worldDifficultyCombo.setCurrentIndex(self.worldMeta.Difficulty)
+        self.generationTypeCombo.setCurrentIndex(self.GENERATOR_TYPES.index(self.worldMeta.generatorName))
+
+        self.worldNameLineEdit.setText(self.worldMeta.LevelName)
+        self.worldNameLineEdit.setText(self.worldMeta.LevelName)
+        self.generatorSeedLineEdit.setText(unicode(str(self.worldMeta.RandomSeed), encoding='utf-8'))
+        self.generatorOptionsLineEdit.setText(self.worldMeta.generatorOptions)
+
+        self.spawnX.setValue(self.worldMeta.SpawnX)
+        self.spawnY.setValue(self.worldMeta.SpawnY)
+        self.spawnZ.setValue(self.worldMeta.SpawnZ)
+
+        time = self.worldMeta.DayTime + 30000  # On time = 0, it's day 1, 6:00. Day 0, 0:00 is -30000
+        day = time / 24000
+        hour = (time % 24000) / 1000
+        minute = int((time % 1000) / (1000.0 / 60.0))
+
+        self.timeDays.setValue(day)
+        self.timeHours.setValue(hour)
+        self.timeMinutes.setValue(minute)
+
+        self.lockedDifficultyBool.setChecked(bool(self.worldMeta.DifficultyLocked))
+        self.commandsBool.setChecked(bool(self.worldMeta.allowCommands))
+        self.hardcoreBool.setChecked(bool(self.worldMeta.hardcore))
+
+    def setupPanel(self):
+        self.defaultGamemodeCombo.clear
+        self.defaultGamemodeCombo.addItems(self.GAMEMODES)
+        self.defaultGamemodeCombo.currentIndexChanged.connect(self.defaultGamemodeChanged)
+
+        self.worldDifficultyCombo.clear
+        self.worldDifficultyCombo.addItems(self.DIFFICULTIES)
+        self.worldDifficultyCombo.currentIndexChanged.connect(self.worldDifficultyChanged)
+
+        self.generationTypeCombo.clear
+        self.generationTypeCombo.addItems(self.GENERATOR_TYPES)
+        self.generationTypeCombo.currentIndexChanged.connect(self.generationTypeChanged)
+
+        self.worldNameLineEdit.editingFinished.connect(self.worldNameChanged)
+        self.generatorSeedLineEdit.editingFinished.connect(self.seedChanged)
+        self.generatorOptionsLineEdit.editingFinished.connect(self.generatorOptionsChanged)
+
+        self.spawnX.editingFinished.connect(self.spawnXChanged)
+        self.spawnY.editingFinished.connect(self.spawnYChanged)
+        self.spawnZ.editingFinished.connect(self.spawnZChanged)
+
+        self.timeDays.editingFinished.connect(self.timeChanged)
+        self.timeHours.editingFinished.connect(self.timeChanged)
+        self.timeMinutes.editingFinished.connect(self.timeChanged)
```

Comment:
I don't imagine anyone cares about the exact time of day.

This is what I would do: Combine hours and minutes into a single field. Next to that field, put a QSlider with icons above it for midnight, dawn, noon, dusk, midnight again. (I don't remember if you can put the icons right into QSlider.)